### PR TITLE
HPCC-14405 Redis Plugin - cache all subscription connections

### PR DESCRIPTION
**Pull request by richardkchapman**
Fix build error on CLang

Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@jamienoss FYI
@ghalliday Please review/merge

File: plugins/redis/redis.cpp [hunk 1]
```diff
@@ -354,7 +354,7 @@ void * Connection::redisCommand(const char * format, ...)
 }
 int Connection::setTimeout(unsigned _timeout)
 {
-    struct timeval to = { _timeout/1000, (_timeout%1000)*1000 };
+    struct timeval to = { (int) _timeout/1000, (int) (_timeout%1000)*1000 };
```
Comment (ghalliday) [inline plugins/redis/redis.cpp]: I think this should really be

{ (time_t)(_timeout/1000), (suseconds_t)((_timeout%1000)*1000) };

something similar above (and don't change the type of timeStillLeft)

**Comment by ghalliday** [comment]
@richardkchapman the types on other platforms might be unsigned, and start generating warnings/errors there.

File: plugins/redis/redis.cpp (hunk 1 above)
Comment (richardkchapman) [inline plugins/redis/redis.cpp]: Getting a sense of deja vu here! I think we had a similar discussion last time I fixed this...

**Pull request merged**